app/Models/Context.php

```diff
@@ -55,22 +55,23 @@ class FreshRSS_Context {
 	/**
 	 * Initialize the context for the global system.
 	 */
-	public static function initSystem($reload = false) {
-		if ($reload || FreshRSS_Context::$system_conf == null) {
+	public static function initSystem($reload = false): ?\FreshRSS_SystemConfiguration {
+		if ($reload || self::$system_conf == null) {
 			//TODO: Keep in session what we need instead of always reloading from disk
-			FreshRSS_Context::$system_conf = FreshRSS_SystemConfiguration::init(DATA_PATH . '/config.php', FRESHRSS_PATH . '/config.default.php');
+			self::$system_conf = FreshRSS_SystemConfiguration::init(DATA_PATH . '/config.php', FRESHRSS_PATH . '/config.default.php');
```

Comment:
Any advantage of using self instead of the class name here?
If find it more clear to use the same construct (for public members) everywhere.

Reply:
There are several advantages:

- reduced cognitive load because we know unambiguously that we are calling the current class and not an external class
- performance, the php parser doesn't have to go through the autoload again because it is explicitly told that the reference is to itself.
- If you change the name of the class, you don't have to go through all the calls again, at the risk of forgetting some and introducing a bug

Reply:
If you are able to find a reference for the performance difference, I would be interested.
For the other two points, I find the opposite (lower cognitive load because it is the same everywhere, and easier to check in the whole codebase in case of changes), so that seems to be more a matter of personal preference (and please refrain from changing things that in a category of pure personal preference, as we can otherwise spend all our time on that, it risks introducing bugs, and it pollutes the git history).

Reply:
By using self instead of the current class name, you can ensure that you're calling a method or static property of the class you're in, rather than an inherited or redefined class. This avoids ambiguity and ensures that the expected behavior is executed. In addition, it makes the code clearer and more maintainable by explicitly stating what a static method belongs to.

Is not personal preference
I had read that the performance was improved, I can't find the article anymore but as soon as I do I'll share it with you

Reply:
`self::` can also be used to call inherited methods. The rest of what you say can be seen both ways. So while awaiting more evidence on the subject, I would still leave it to personal preference.

Reply:
last little question, why use self:: for properties but not for methods?
cf lines : 325 to 337

Reply:
Written by different persons and different points in time (some lines are 8 years old, I can see), so a bit of inconsistency in personal preferences, which is inconsequential. I would rather not update such lines except if there is a compelling reason to do so, as I give a higher value to a lean git history and to approaches limiting edit wars (in other words, in terms of personal preferences, the original author of those lines should mostly win).

One case see that one of the lines is mine, and I have kept the style of the surrounding lines:
![image](https://user-images.githubusercontent.com/1008324/216356122-986cf43a-702b-4731-b6b9-12b1ce12afc5.png)